sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/LiveResourceManager.cs

```diff
@@ -30,241 +30,281 @@ public sealed class LiveResourceManager
+        /// <returns>The newly created Event Hub.</returns>
         ///
-        private static bool IsRetriableStatus(HttpStatusCode statusCode) =>
-            ((statusCode == HttpStatusCode.Unauthorized)
-                || (statusCode == ((HttpStatusCode)408))
-                || (statusCode == HttpStatusCode.Conflict)
-                || (statusCode == ((HttpStatusCode)429))
-                || (statusCode == HttpStatusCode.InternalServerError)
-                || (statusCode == HttpStatusCode.ServiceUnavailable)
-                || (statusCode == HttpStatusCode.GatewayTimeout));
+        public async Task<EventHubResource> CreateEventHubAsync(string eventHubName,
+                                                                int partitionCount,
+                                                                IList<string> consumerGroups = default,
+                                                                CancellationToken cancellationToken = default)
+        {
+            var ehNamespace = ArmClient.GetEventHubsNamespaceResource(
+                EventHubsNamespaceResource.CreateResourceIdentifier(
+                    EventHubsTestEnvironment.Instance.SubscriptionId,
+                    EventHubsTestEnvironment.Instance.ResourceGroup,
+                    EventHubsTestEnvironment.Instance.EventHubsNamespace));
 
-        /// <summary>
-        ///   Determines whether the specified exception is considered eligible to retry the associated
-        ///   operation.
-        /// </summary>
-        ///
-        /// <param name="ex">The exception to consider.</param>
-        ///
-        /// <returns><c>true</c> if the exception is eligible for retries; otherwise, <c>false</c>.</returns>
-        ///
-        private static bool ShouldRetry(Exception ex) =>
-            ((IsRetriableException(ex)) || (IsRetriableException(ex?.InnerException)));
+            var eventHubConfig = new EventHubData { PartitionCount = partitionCount, MessageRetentionInDays = 1 };
+            var eventHub = await ehNamespace.GetEventHubs().CreateOrUpdateAsync(WaitUntil.Completed, eventHubName, eventHubConfig).ConfigureAwait(false);
 
-        /// <summary>
-        ///   Determines whether the type of the specified exception is considered eligible to retry
-        ///   the associated operation.
-        /// </summary>
-        ///
-        /// <param name="ex">The exception to consider.</param>
-        ///
-        /// <returns><c>true</c> if the exception type is eligible for retries; otherwise, <c>false</c>.</returns>
-        ///
-        private static bool IsRetriableException(Exception ex)
-        {
-            if (ex == null)
-            {
-                return false;
-            }
+            var groups = consumerGroups ?? new List<string>();
+            var consumerGroupCollection = eventHub.Value.GetEventHubsConsumerGroups();
 
-            switch (ex)
+            if ((groups != default) && (groups.Count > 0))
```

Comment:
nit - I think you can simplify to: (may need a ` ?? false` at the end)
```suggestion
            if (groups?.Count > 0)
```